### PR TITLE
Fixup let SIMD subpackage be unconditionally enabled since C++17 is now required

### PR DESCRIPTION
**Pull request by dalg24**

### CODE REVIEW

**Review by masterleinad — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -193,14 +193,8 @@ KOKKOS_SETUP_BUILD_ENVIRONMENT()
 
 OPTION(BUILD_SHARED_LIBS "Build shared libraries" OFF)
 
-SET(KOKKOS_EXT_LIBRARIES Kokkos::kokkos Kokkos::kokkoscore Kokkos::kokkoscontainers Kokkos::kokkosalgorithms)
```
Comment (crtrott) [inline CMakeLists.txt:196]: why is this gone?

**Review by crtrott — CHANGES_REQUESTED** [review]
Why is KOKKOS_EXT_LIBRARIES now gone?

File: CMakeLists.txt (hunk 1 above)
Comment (crtrott) [reply CMakeLists.txt:196]: what was this used for at all?

Comment (dalg24) [reply CMakeLists.txt:196]: Never used (I checked in Trilinos too)

Comment (dalg24) [reply CMakeLists.txt:196]: Also gripped for `_EXT_LIBRARIES` and packages don't define it

Comment (Rombur) [reply CMakeLists.txt:196]: It is used [here](https://github.com/dalg24/kokkos/blob/develop/cmake/KokkosConfig.cmake.in#L12)

Comment (Rombur) [reply CMakeLists.txt:196]: Sorry it was from an old commit. We don't use this anymore

**Review by ibaned — APPROVED** [review]

**Review by crtrott — APPROVED** [review]

**Pull request merged**